smpp/pdu/pdufield/tlvtypes.go

```diff
@@ -0,0 +1,586 @@
+package pdufield
+
+import (
+	"reflect"
+)
+
+func GetParameters(v interface{}) map[TLVType]string {
+	m := make(map[TLVType]string)
+	val := reflect.ValueOf(v).Elem()
+	for i := 0; i < val.NumField(); i++ {
+		valueField := val.Field(i)
+		typeField := val.Type().Field(i)
+		key := valueField.Interface().(TLVType)
+		value := typeField.Name
+		m[key] = value
+	}
+	return m
+}
```

Comment:
If the matching TLV fields allowed in a PDU were stored in a `map` rather than a struct, iterating it would be much easier.